### PR TITLE
Fix various issue with the spectral index module

### PR DESCRIPTION
**Pull request by darafferty**
This PR fixes a number of issues in the spectral index module, most of which were related to the the lack of channel-specific masking. An option was also added to allow the user to specify channels to flag before spectral indices are calculated. Fixes issue #201.

### CODE REVIEW

File: bdsf/spectralindex.py [hunk 1]
```diff
@@ -60,7 +60,7 @@ def __call__(self, img):
                 unav_freqs = freqin[good_chans]
                 nmax_to_avg = img.opts.specind_maxchan
                 nchan = unav_image.shape[0]
-                mylog.info('After initial flagging of channels by rms, %i good channels remain' % (nchan,))
+                mylogger.userinfo(mylog, 'Number of channels remaining after initial flagging', str(nchan))
```
Comment (gmloose) [inline bdsf/spectralindex.py:63]: Curious as to why you changed this from `mylog.info` to `mylogger.userinfo`.

**Review by gmloose — APPROVED** [review]
You might want to run your sources through `pylint` and at least fix all warnings.

File: bdsf/spectralindex.py (hunk 1 above)
Comment (darafferty) [reply bdsf/spectralindex.py:63]: That was so that the message is printed to the terminal and not only to the log file.

**Comment by darafferty** [comment]
I fixed the `pylint` warnings -- thanks for suggesting that.

**Pull request merged**